networkx/algorithms/tests/test_min_edge_cover.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python
+from nose.tools import *
+import networkx as nx
+
+
+def test_trivial1():
+    """Empty graph"""
+    G = nx.Graph()
+    assert_equal(nx.min_edge_cover(G), {})
+
+
+def test_trivial2():
+    """ loop"""
+    G = nx.Graph()
+    G.add_edge(0, 0, weight=100)
+    assert_equal(nx.min_edge_cover(G), {0: [0, 0]})
+
+
+def test_trivial3():
+    """Single edge"""
+    G = nx.Graph()
+    G.add_edge(0, 1)
+    assert_equal(nx.min_edge_cover(G),
+                 {0: [1], 1: [0]})
+
+
+def test_bipartite():
+    """Bipartite Graph"""
+    G = nx.Graph()
+    G.add_edges_from([(1, 2), (3, 4), (5, 6), (1, 3)])
+    assert_equal(nx.min_edge_cover(G),
+                 {1: [2], 2: [1], 3: [4], 4: [3], 5: [6], 6: [5]})
+
+
+def test_not_exist():
+    """Min Cover Does not exist"""
+    G = nx.Graph()
+    G.add_edges_from([(1, 2), (3, 4), (5, 6)])
+    G.add_node(7)
+    # no edge from 7
+    assert_equal(nx.min_edge_cover(G), {})
+
+
+def test_complete_graph():
```

Comment:
There is a problem with this sort of test. You are relying on an arbitrary ordering of the nodes and/or edges. However, you don't know that the `maximum_matching` function will always return this matching. It would be better to write a function `is_edge_covering()` (perhaps even in the `covering` module, exposing it as part of the public API) that decides whether a set of edges is a valid edge covering. Then call that method here, like `assert_true(is_edge_covering(G, covering))`.